gnosis/gp-swap-ui
# [407] Unsupported Token UX

Closes #407

Main points:
1. leverages existing Uni unsupported Token logic blocking trade based on `isSwapSupported` flag in swapMod
2. adds custom state for dynamically set unsupportedTokens by address map
3. does not create expensive TokenList object as Uni does for tokens and instead simply checks token address against unsupportedTokens map and creates expensive Token object if necessary
4. useRefetchFeeCallback, which fetches fee/price for a swap, now checks server response for 400 (unsupported token) response and sets redux state with unsupported token

## To test:
1. mainnet
2. add KICK token: 0xc12d1c73ee7dc3615ba4e37e4abfdbddfa38907e
3. HOGE also works: 0xfad45e47083e4607302aa43c65fb3106f1cd7607
4. should not be tradable
5. in currency list u should see the Unsupported flag on token

## Screenshots
<img width="301" alt="Screenshot 2021-05-11 at 13 03 34" src="https://user-images.githubusercontent.com/21335563/117812728-12f85f00-b25a-11eb-9cf0-485313169448.png">
<img width="301" alt="Screenshot 2021-05-11 at 13 03 44" src="https://user-images.githubusercontent.com/21335563/117812736-155ab900-b25a-11eb-9197-553509b53e4f.png">
